test/test_runner/base_test_runner.go

```diff
@@ -84,9 +97,27 @@ func (t *TestRunner) runAgent() (status.TestGroupResult, error) {
 	}
 
 	agentConfigPath := filepath.Join(agentConfigDirectory, t.TestRunner.GetAgentConfigFileName())
-	log.Printf("Starting agent using agent config file %s", agentConfigPath)
-	common.CopyFile(agentConfigPath, configOutputPath)
-	err = common.StartAgent(configOutputPath, false)
+
+	if t.TestRunner.SSM() {
+		log.Printf("Starting agent from ssm parameter %s", agentConfigPath)
+		agentConfigByteArray, err := os.ReadFile(agentConfigPath)
+		if err != nil {
+			return testGroupResult, errors.New("failed while reading config file")
+		}
+
+		agentConfig := string(agentConfigByteArray)
+
+		err = awsservice.PutStringParameter(t.TestRunner.SSMParameterName(), agentConfig)
+		if err != nil {
+			return testGroupResult, errors.New("failed to upload ssm parameter")
+		}
```

Comment:
Why not just put this in the specific test runner's `SetupBeforeAgentRun()` function